aptos-move/aptos-aggregator/src/aggregator_extension.rs

```diff
@@ -302,10 +302,29 @@ impl AggregatorData {
     /// delta state.
     /// Note: when we say "aggregator instance" here we refer to Rust struct and
     /// not to the Move aggregator.
-    pub fn get_aggregator(&mut self, id: AggregatorID, limit: u128) -> &mut Aggregator {
+    pub fn get_aggregator(
+        &mut self,
+        id: AggregatorID,
+        limit: u128,
+        resolver: &dyn TableResolver,
+        is_aggregator_enabled: bool,
+    ) -> &mut Aggregator {
+        let (state, value) = if is_aggregator_enabled {
+            (AggregatorState::PositiveDelta, 0)
+        } else {
+            //If aggregator mode is disabled, then read the aggregator value from the table and initialize with it.
+            let key_bytes = id.key.0.to_vec();
+            resolver
+                .resolve_table_entry(&id.handle, &key_bytes)
+                .map_err(|_| (AggregatorState::PositiveDelta, 0))
```

Comment:
Why this? We should return Err(...) if resolver fails?

Reply:
If the resolver fails, doesn't it mean there is no entry corresponding to that aggregagorId in the table, a we need to create a new aggregator?

Reply:
I think you can call just call `read_and_materialize`?

Reply:
Resolver goes to DB, if it fails something really wrong happened

Reply:
Do you mean, I should first initialize the aggregator with 0 value, and then call `read_and_materialize` if the `is_aggregator_enabled` flag is set to false? If I call `read_and_materialize` when the AggregatorState is Data, it simply returns the current value (which is 0).

Reply:
You can initialise it with a `PositiveDelta`,  and then read will go to storage?

Reply:
Done.